x/staking/keeper/grpc_query_test.go

```diff
@@ -793,13 +793,363 @@ func createValidators(t *testing.T, ctx sdk.Context, app *simapp.SimApp, powers
 	app.StakingKeeper.SetNewValidatorByPowerIndex(ctx, val1)
 	app.StakingKeeper.SetNewValidatorByPowerIndex(ctx, val2)
 
-	_, err := app.StakingKeeper.Delegate(ctx, addrs[0], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[0]), types.Unbonded, val1, true)
+	err := delegateCoinsFromAccount(ctx, app, addrs[0], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[0]), val1)
 	require.NoError(t, err)
-	_, err = app.StakingKeeper.Delegate(ctx, addrs[1], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[1]), types.Unbonded, val2, true)
+	err = delegateCoinsFromAccount(ctx, app, addrs[1], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[1]), val2)
 	require.NoError(t, err)
-	_, err = app.StakingKeeper.Delegate(ctx, addrs[0], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[2]), types.Unbonded, val2, true)
+	err = delegateCoinsFromAccount(ctx, app, addrs[0], app.StakingKeeper.TokensFromConsensusPower(ctx, powers[2]), val2)
 	require.NoError(t, err)
 	applyValidatorSetUpdates(t, ctx, app.StakingKeeper, -1)
 
 	return addrs, valAddrs, vals
 }
+
+func (suite *KeeperTestSuite) TestGRPCQueryQueuedDelegations() {
+	app, ctx, queryClient, addrs, vals := suite.app, suite.ctx, suite.queryClient, suite.addrs, suite.vals
+	addrAcc := addrs[0]
+	addrVal1 := vals[0].OperatorAddress
+	valAddr, err := sdk.ValAddressFromBech32(addrVal1)
+	suite.NoError(err)
+	coin := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(5))
+
+	delegation := types.MsgDelegate{
+		DelegatorAddress: addrAcc.String(),
+		ValidatorAddress: valAddr.String(),
+		Amount:           coin,
+	}
+	epochNumber := app.StakingKeeper.GetEpochNumber(ctx)
+	app.StakingKeeper.QueueMsgForEpoch(ctx, epochNumber, &delegation)
+	var req *types.QueryQueuedMsgDelegatesRequest
+
+	testCases := []struct {
+		msg      string
+		malleate func()
```

Comment:
Please add a comment explaining the need to wrap what looks like a simple struct literal in a func.